### PR TITLE
Add support for arduino firmata

### PR DESCRIPTION
**Pull request by Chiff**
# Arduino Firmata protocol for pysimCoder

## Prerequisites

Install the [Arduino IDE](https://www.arduino.cc/en/software) and the upload StandardFirmata on your Arduino (Arduino
IDE > File > Examples > Firmata > StandardFirmata).

---

## New Blocks

- setup - Initializes Firmata on `port='/dev/ttyACM0'` with `baud=57600`
- analogWrite - uses arduino analogWrite function on `pin` with `mode` (firmata supported
  modes: https://github.com/Chiff/FirmataC/blob/master/includes/firmata.h#L7-L13)
- digitalWrite - uses arduino digitalWrite function on `pin`
- pinRead - reads pin value from arduino `pin` with `mode` (firmata supported
  modes: https://github.com/Chiff/FirmataC/blob/master/includes/firmata.h#L7-L13)
- customRead - array from 0 to 6
  - customRead[0] - VL53L0X distance sensor - distance in mm - SYSEX format = `START_SYSEX 0x72 msb lsb END_SYSEX`
  - customRead[1] - **unused at the moment*
  - customRead[2] - **unused at the moment*
  - customRead[3] - **unused at the moment*
  - customRead[4] - **unused at the moment*
  - customRead[5] - **unused at the moment*
  - customRead[6] - **unused at the moment*

Helpers:

- help_HeatShield_temperature - customized block to translate voltage to temperature based on the shield
- help_TCLab_temperature - customized block to translate voltage to temperature based on the shield

#### SocketServer

Express server that can be used to communicate with the Arduino.

Creates:

- simple web page to plot data from the Arduino and send commands to the Arduino
- express API
- TCP server that listens on port 5000
- Unix socket client

Prerequisites:

- Node.js (see package.json for version requirements)
- npm or yarn (see package.json for version requirements)

Usage:

- Navigate to pysimCoder/CodeGen/arduinoFirmata/tools/SocketServer
- If necessary edit the `src/config.js` file
- Run `npm run start` or `yarn start`

### CODE REVIEW

**Comment by Chiff** [comment]
tested on: arduino uno, mega, leonardo, nano
diagrams and outputs are located in `pysimCoder\CodeGen\arduinoFirmata\tools\SocketServer\outputs`

File: CodeGen/LinuxRT/devices/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.o
```
Comment (michallenc) [inline CodeGen/LinuxRT/devices/.gitignore:1]: This should be already covered in CodeGen/LinuxRT/.gitignore added in 6d3dba3

Comment (Chiff) [reply CodeGen/LinuxRT/devices/.gitignore]: fixed

**Comment by robertobucher** [comment]
Hi Martin

Do you have some examples to put under

https://github.com/robertobucher/pysimCoder-examples

to try your additions for arduino?
Thanks in advance for your contribution!
Roberto

**Pull request merged**